pubpub/pubpub
# feature: Collection header block

This PR adds a new type of Layout block for Collections which will appear in default Collection layouts and act as a header. There's also a lot of refactoring in here:

- TypeScript definitions for attribution/contributor related utilities (we really need to reconcile these two names)
- New `ContributorAvatars` component to show a list of contributor avatars like in Pub preview blocks. This has a hover popover showing more details about each contributor, which itself was factored out of the Pub Header details pane.

Feedback on design welcome!

_A fully loaded Collection header block:_
![image](https://user-images.githubusercontent.com/2208769/97047477-08d57400-1547-11eb-9e48-2c4dffbc6701.png)

_Test plan:_
- Check that Collection header blocks can be added, modified, and removed in the Collection layout editor. Check that they do not appear in the Page layout editor.
- Check these out on the reader side (and in mobile view)
- Test that new Collections have a header block
- Make sure avatars in PubPreviews are still okay (refactored to use shared `ContributorAvatars`)
- Make sure Pub Header details pane contributors list is still okay (and on mobile) (refactored to use `ContributorsListing`)
